### PR TITLE
Discord-Facing Module Framework

### PR DESCRIPTION
**Pull request by SolarysDev**
After some revision from the last PR and some feedback from the Discord, the new Discord-facing module framework is ready to be implemented.

### CODE REVIEW

File: src/Volvox.Helios.Core/Modules/DiscordFacing/DiscordFacingContext.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+{
+    public class DiscordFacingContext
+    {
+        public SocketUserMessage Message { get; }
+        public ISocketMessageChannel Channel { get; }
+        public SocketUser User { get; }
+        public DiscordSocketClient Client { get; }
+        public string GivenPrefix { get; }
+
+        public DiscordFacingContext(SocketUserMessage m, DiscordSocketClient c, string p)
```
Comment (BillChirico) [inline src/Volvox.Helios.Core/Modules/DiscordFacing/DiscordFacingContext.cs]: Can we add comments on these properties as they are not directly known what they do just based on name

File: src/Volvox.Helios.Core/Modules/DiscordFacing/Commands/DExampleModule.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        public async Task ExecuteAsync(DiscordFacingContext context)
+        {
+            await context.Channel.SendMessageAsync("no");
+        }
+
+        public async Task<bool> TryTrigger(DiscordFacingContext context)
+        {
+            if (context.Message.Content.StartsWith($"{context.GivenPrefix}help"))
+            {
+                await ExecuteAsync(context);
```
Comment (BillChirico) [inline src/Volvox.Helios.Core/Modules/DiscordFacing/Commands/DExampleModule.cs]: Should the manager call the Execute if the Trigger passes?

Comment (SolarysDev) [reply src/Volvox.Helios.Core/Modules/DiscordFacing/Commands/DExampleModule.cs]: Might be better to let ITriggerable handle that, as an ITriggerable knows what it's going to do if the check passes based on the module implementing it, while the manager doesn't. I also designed ITriggerable with the intent of it possibly being used to act as a check for other types of modules. That's why it's in Common and not DiscordFacing.

File: src/Volvox.Helios.Core/Modules/DiscordFacing/DiscordFacingContext.cs (hunk 1 above)
Comment (SolarysDev) [reply src/Volvox.Helios.Core/Modules/DiscordFacing/DiscordFacingContext.cs]: Will fix as soon as possible

File: src/Volvox.Helios.Core/Modules/DiscordFacing/Commands/DExampleModule.cs (hunk 2 above)
Comment (BillChirico) [reply src/Volvox.Helios.Core/Modules/DiscordFacing/Commands/DExampleModule.cs]: Gotcha, makes sense.

File: src/Volvox.Helios.Core/Modules/DiscordFacing/DiscordFacingContext.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        /// Fed by DiscordFacingManager
+        /// </summary>
+        public DiscordSocketClient Client { get; }
+        
+        /// <summary>
+        /// The prefix the invoking guiild has set for the bot. Useful once Bapes finishes the settings framework.
+        /// </summary>
+        public string GivenPrefix { get; }
+
+        public DiscordFacingContext(SocketUserMessage m, DiscordSocketClient c, string p)
```
Comment (BillChirico) [inline src/Volvox.Helios.Core/Modules/DiscordFacing/DiscordFacingContext.cs]: Could you add parameter comments here as they will be most important when creating a new `DiscordFacingContext`

Comment (SolarysDev) [reply src/Volvox.Helios.Core/Modules/DiscordFacing/DiscordFacingContext.cs]: Done

**Comment by SolarysDev** [comment]
[DModule Framework](https://trello.com/c/G2gMESQI/1-dmodule-framework)

**Comment by BillChirico** [comment]
Need changes

**Pull request closed without merging**